### ISSUE TITLE
Always open new tab outside of tab stack?

### ISSUE THREAD

**Issue by reallyuniquename** [yfdyh000/tabutils#93]
Is there any way to force new tab to open outside of the current tab stack? 

E.g.: current (selected) tab is inside the stack, I open new tab and it doesn't autostack.

**Comment by raphalelol** [comment]
yes i have this problem too, and when setting "open new tab next to current tab" TO "all but new tab" which is suppused to open new tabs outsite of the stack, it doesn't work!

**Comment by yfdyh000** [comment]
Set "open new tab next to current tab to "All" is ok?
I'm not sure I understand.

**Comment by raphalelol** [comment]
open new tabs next to current

 SET TO ( all ) === working

 SET TO ( all but new tab ) == not working 

 SET TO ( none but links ) == working

SET TO ( none ) == working

we want the second option to work, it is what we mean new tab outside of the stack

**Comment by yfdyh000** [comment]
I see, but may not understand yours needs. The "open new tab next to current tab" option for new tab only work when set to "all".
You need an option for open new tab to inside of current tab stack?

**Comment by raphalelol** [comment]
no i don't want (open new tabs next to current tab) to work for new tabs
i want to work for all tabs EXCEPT new tabs

this may help

i have
[tab1][tab2][tab3]

**suppose i set it to ( all but new tabs)**
**suppose current tab is [tab2]**

so when i open new tab, i expect this:

[tab1][tab2][tab3][newtab]

_which what the option ( all but new tabs) is supposed to do._
and which is I and the OP want

BUT the problem is we get this

[tab1][tab2][newtab][tab3]

 by the way, your addon is great, it's simply the best and lightest compared to others, and this problem is very minor and doesn't effect my experience.

**Comment by yfdyh000** [comment]
Many thanks, I understand it now, but it is strange that it works as you expect for me, I could not reproduce it for now. Fx47 or Fx48 on Win10, tab_utilities_fixed-1.5.2016.06.08-fx.xpi.

**Comment by reallyuniquename** [comment]
Right now it works this way.

I have a tab stack and TAB2 is current (selected) tab: **[**tab1**|**_TAB2_**|**tab3**]**

Then I open a new tab and it becomes: **[**tab1**|**tab2**|**_NEW_TAB_**|**tab3**]**

I'd like an option to force new tab to spawn AFTER the tab stack like this: **[**tab1**|**tab2**|**tab3**]**_NEW_TAB_

In short I'd like to open a new tab next to current one EXCEPT IF current one is inside that stack (then it should be opened AFTER the tab stack).

Is there any way you can implement this?

**Comment by yfdyh000** [comment]
These are two different needs?

I don't have an idea to improve it.

**Comment by reallyuniquename** [comment]
Yeah, @raphalelol request is slightly different. 

> I don't have an idea to improve it.

Isn't it possible to detect if current tab is inside the stack (has "group" attribute) and then open new tab AFTER the stack (after the tab with "group-last" attribute)?

**Comment by yfdyh000** [comment]
The relevant code: https://github.com/yfdyh000/tabutils/blob/abc592071f174c02eb13bb78a8bb7a5cc8e97405/chrome/content/tabutils.js#L679

I'm not truly understand the code and needs, so it is difficult to achieve.
In addition, adding an option may be more difficult for me.

**Comment by reallyuniquename** [comment]
Dirty solution to open all new tabs outside of the stack in case someone needs it. 

Line 656 in `tabutils.js`, replace 

``` javascript
function shouldStack(tab) { let args = tab.arguments; return args.aReferrerURI || args.aRelatedToCurrent && args.aURI != "about:blank"; }
```

 with 

``` javascript
function shouldStack(tab) { return true; }
```

New tab will be opened unstacked on the right side of the current stack.

@raphalelol I can't reproduce your issue, `all but new tab` works as intended.
